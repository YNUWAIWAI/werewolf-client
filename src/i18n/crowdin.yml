project_identifier_env: CROWDIN_PROJECT_ID
api_key_env: CROWDIN_API_KEY
base_path : "./"
preserve_hierarchy: true

files:
  -
    source: '*/en.json'
    translation: '%original_path%/%locale%.json'
    languages_mapping:
      locale:
        'af': 'af'
        'ar': 'ar'
        'bs-BA': 'bs-BA'
        'ca': 'ca'
        'cs': 'cs'
        'da': 'da'
        'de': 'de'
        'el': 'el'
        'es-ES': 'es-ES'
        'fa': 'fa-IR'
        'fi': 'fi'
        'fr': 'fr'
        'he': 'he'
        'hu': 'hu'
        'id': 'id-ID'
        'it': 'it'
        'ja': 'ja'
        'ko': 'ko'
        'mr': 'mr-IN'
        'nl': 'nl'
        'no': 'no-NO'
        'pl': 'pl'
        'pt-BR': 'pt-BR'
        'pt-PT': 'pt-PT'
        'ro': 'ro'
        'ru': 'ru'
        'sk': 'sk-SK'
        'sr': 'sr'
        'sv-SE': 'sv-SE'
        'tr': 'tr'
        'uk': 'uk'
        'vi': 'vi'
        'zh-CN': 'zh-CN'
        'zh-TW': 'zh-TW'
