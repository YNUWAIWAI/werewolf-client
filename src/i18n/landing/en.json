{
  "lead-flavor": "* Flavor text *",
  "lead-title": "LiCOS: A Game Of Werewolves",
  "login-error": "Invalid email or password",
  "login-label-email": "Email address",
  "login-label-password": "Password",
  "login-navigation": "Login",
  "login-submit": "Login",
  "signup-confirmation": "Please agree to our <a>Terms, Data Policy and Cookies Policy</a> before signing up.",
  "signup-error": "Invalid name, email or password",
  "signup-label-email": "Email address",
  "signup-label-name": "Name",
  "signup-label-password": "Password",
  "signup-navigation": "Signup",
  "signup-submit": "Signup",
  "Description.human-vs-ai": "Human vs AI.",
  "Description.human-vs-werewolves": "Humans vs Werewolves.",
  "Description.survive": "Survive in the post-truth world.",
  "Description.1": "This is a game about a village beset by werewolves. Most players will be ordinary villagers, but some of them will be werewolves, and other ones may have special abilities. In the village, werewolves attacks a villager every night. Ordinary villagers can not distinguish werewolves from villagers. To eliminate werewolves, they decide that, every day, they put a suspicious villager to death at noon. If all werewolves are put to death, the next morning nobody will be dead.",
  "Description.2": "Using your web browser, you can directly participate in the game as a human player.",
  "Description.3": "If you want more of a challenge, you can develop your own robot and make it participate as a robot player.",
  "Description.button.how-to-play": "How to play",
  "Description.button.how-to-develop-a-robot": "How to develop a robot",
  "Sponsor.sponsoredBy": "Sponsored By",
  "UpdateNotes.header": "Update notes"
}
