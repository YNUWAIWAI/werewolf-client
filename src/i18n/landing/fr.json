{
  "lead-flavor": "* Flavor text *",
  "lead-title": "LiCOS: A Game Of Werewolves",
  "login-error": "Invalid email or password",
  "login-label-email": "Email",
  "login-label-password": "Password",
  "login-navigation": "Login",
  "login-submit": "Login",
  "signup-confirmation": "Please agree to our <a>Terms, Data Policy and Cookies Policy</a> before signing up.",
  "signup-error": "Invalid name, email or password",
  "signup-label-email": "Email",
  "signup-label-name": "Name",
  "signup-label-password": "Password",
  "signup-navigation": "Signup",
  "signup-submit": "Signup"
}
