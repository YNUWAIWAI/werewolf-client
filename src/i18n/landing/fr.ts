export default {
  'login-error': 'Invalid email or password',
  'login-label-email': 'Email',
  'login-label-password': 'Password',
  'login-submit': 'Login',
  // 'signup-confirmation': '',
  'signup-email': 'Email',
  'signup-error': 'Invalid name, email or password',
  'signup-name': 'Name',
  'signup-password': 'Password',
  'signup-submit': 'Signup'
}
