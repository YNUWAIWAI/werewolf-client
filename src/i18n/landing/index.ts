import enMessages from './en'
import frMessages from './fr'
import itMessages from './it'
import jaMessages from './ja'

export const enum Language {
  en = 'en',
  fr = 'fr',
  it = 'it',
  ja = 'ja'
}
interface Messages {
  [id: string]: string
}

export const getMessages = (locale: Language): Messages => {
  switch (locale) {
    case 'en':
      return enMessages
    case 'fr':
      return frMessages
    case 'it':
      return itMessages
    case 'ja':
      return jaMessages
    default:
      throw Error(`${locale} is unsuported`)
  }
}
