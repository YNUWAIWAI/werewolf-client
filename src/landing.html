@import helper._
@import models.UserModel
@(loginForm: Form[UserModel], signupForm: Form[UserModel])(implicit request: RequestHeader, messagesProvider: MessagesProvider)
@template("Home", "landing.css") {
  <div class="lead">
    <div class="title">
      <p>Werewolf: The Game</p>
    </div>
    <div class="flavor">
      <p>* Flavor text *</p>
    </div>
    <div class="image">
      <svg viewBox="0 0 100 100" xmlns="http://www.w3.org/2000/svg">
        <circle cx="50" cy="50" r="50"/>
        <text x="12.5" y="75" fill="white">çµµ</text>
      </svg>
    </div>
  </div>
  @form(action = routes.AuthController.loginResult(), 'class->"login") {
    @if(loginForm.hasErrors) {
      <p class="error">
        Invalid email or password
      </p>
    }
    <div class="field">
      <dl id="login-email_field">
        <dt><label class="label" for="login-email">Email</label></dt>
        <dd><input type="text" name="email" id="login-email" class="input" value="" autocomplete="off" autofocus required></dd>
      </dl>
    </div>
    <div class="field">
      <dl id="login-divassword_field">
        <dt><label class="label" for="login-password">Password</label></dt>
        <dd><input type="password" name="password" id="login-password" class="input" value="" autocomplete="off" minLength="8" required></dd>
      </dl>
    </div>
    @CSRF.formField
    <input class="button" type="submit" value="Login">
  }
  @form(action = routes.AuthController.signupResult(), 'class->"signup") {
    @if(signupForm.hasErrors) {
      <p class="error">
        Invalid name, email or password
      </p>
    }
    <div class="field">
      <dl id="signup-name_field">
        <dt><label class="label" for="signup-name">Name</label></dt>
        <dd><input type="text" name="name" id="signup-name" class="input" value="" autocomplete="off" required></dd>
      </dl>
    </div>
    <div class="field">
      <dl id="signup-email_field">
        <dt><label class="label" for="signup-email">Email</label></dt>
        <dd><input type="text" name="email" id="signup-email" class="input" value="" autocomplete="off" required></dd>
      </dl>
    </div>
    <div class="field">
      <dl id="signup-password_field">
        <dt><label class="label" for="signup-password">Password</label></dt>
        <dd><input type="password" name="password" id="signup-password" class="input" value="" autocomplete="off" minLength="8" required></dd>
      </dl>
    </div>
    @CSRF.formField
    <p class="confirmation">
      Please agree to our <a href="terms" target="_blank">Terms, Data Policy and Cookies Policy</a> before signing up.
    </p>
    <input class="button" type="submit" value="Signup">
  }
}
