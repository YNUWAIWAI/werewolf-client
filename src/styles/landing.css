:root {
  --active-color: rgba(0, 0, 0, 0.07);
  --disabled-color: rgba(0, 0, 0, 0.3);
  --hover-color: rgba(0, 0, 0, 0.03);
  --regular-color: rgba(0, 0, 0, 0);
  --error-color: red;
}

body {
  font-family: 'Noto Sans Japanese', 'ＭＳ ゴシック', 'Osaka-Mono', monospace, sans-serif;
}

.la {
  &--button {
    appearance: none;
    background: var(--regular-color);
    border-radius: 5px;
    cursor: pointer;
    font-size: 1.25rem;
    grid-area: button;
    line-height: 2;

    &:hover {
      background: var(--hover-color);

      &:active {
        background: var(--active-color);
        box-shadow: inset 2px 2px 1px 0 grey;
      }
    }
  }

  &--confirmation {
    margin-bottom: 0.5rem;
  }

  &--error {
    color: var(--error-color);
  }

  &--field {
    margin-bottom: 1rem;
  }

  &--flavor {
    margin-top: 10vh;

    & > p {
      font-size: 2rem;
      margin: 1rem;
      text-align: center;
    }
  }

  &--image {
    height: 50%;
    margin: 0 auto;
    width: 50%;
  }

  &--input {
    box-sizing: border-box;
    display: block;
    font-size: 1rem;
    padding: 0.5rem;
    width: 100%;
  }

  &--label {
    display: block;
    margin-bottom: 0.5rem;
  }

  &--landing {
    display: grid;
    grid-template-areas:
      'lead - button'
      'lead form form';
    grid-template-columns: 3fr 1fr 1fr;
    grid-template-rows: 4rem 1fr;
  }

  &--lead {
    grid-area: lead;
    height: 100vh;
    position: relative;
  }

  &--form {
    border: solid 1px black;
    border-radius: 1rem;
    grid-area: form;
    margin: 1rem 1rem auto;
    padding: 1rem;
  }

  &--mode-button {
    background: rgb(156, 204, 101);
    border-radius: 0 0 0 10px;
    color: white;
    font-size: 2rem;
    grid-area: button;
    line-height: 1.5;
    padding: 0.5rem;
  }

  &--title {
    margin-top: 10vh;

    & > p {
      font-size: 3rem;
      margin: 1rem;
      text-align: center;
    }
  }
}
