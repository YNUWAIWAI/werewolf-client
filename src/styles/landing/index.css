.la {
  --min-height--window: 536px;
  --min-width--window: 900px;
  --active-color: rgba(0, 0, 0, 0.07);
  --disabled-color: rgba(0, 0, 0, 0.3);
  --hover-color: rgba(0, 0, 0, 0.03);
  --regular-color: rgba(0, 0, 0, 0);
  --error-color: red;
  --color--background: hsla(0, 0%, 10%);
  --color--background--form: hsla(0, 0%, 15%);
  --color--background--button: hsla(0, 0%, 25%);
  --color--background--button--disabled: hsla(0, 0%, 15%);
  --color--background--button--hover: hsla(0, 0%, 40%);
  --color--border: hsl(0, 0%, 80%);
  --color--font--black: hsl(0, 0%, 10%);
  --color--font--gray: hsl(0, 0%, 50%);
  --color--font--white: hsl(0, 0%, 80%);
  background: var(--color--background);
  color: var(--color--font--white);
  font-family: 'Noto Sans JP', 'ＭＳ ゴシック', 'Osaka-Mono', monospace, sans-serif;

  &--button {
    appearance: none;
    background: var(--color--background--button);
    border: 1px solid var(--color--border);
    border-radius: 5px;
    color: var(--color--font--white);
    cursor: pointer;
    font-size: 1.25rem;
    grid-area: button;
    line-height: 2;

    &:hover {
      background: var(--color--background--button--hover);
    }
  }

  &--confirmation {
    margin-bottom: 0.5rem;
  }

  &--error {
    color: var(--error-color);
  }

  &--field {
    margin-bottom: 1rem;
  }

  &--flavor {
    margin-top: 10vh;

    & > p {
      font-size: 2rem;
      margin: 1rem;
      text-align: center;
    }
  }

  &--trailer {
    border: 0;
    display: block;
    margin: 0 auto;
  }

  &--input {
    appearance: none;
    border: 1px solid rgb(223, 223, 223);
    box-sizing: border-box;
    display: block;
    font-size: 1rem;
    padding: 0.5rem;
    width: 100%;
  }

  &--label {
    display: block;
    margin-bottom: 0.5rem;
  }

  &--grid {
    display: grid;
    grid-template:
      'lead login signup'
      'lead form form'
      'lead notes notes'
      'lead assessment assessment'
      'sponsor sponsor sponsor'
      'footer footer footer'
      / minmax(calc(var(--min-width--window) * 0.6), 3fr) 1fr 1fr;
  }

  &--lead {
    grid-area: lead;
    position: relative;
  }

  &--form {
    background: var(--color--background--form);
    border: solid 1px var(--color--border);
    border-radius: 0 0 4px 4px;
    border-top: none;
    grid-area: form;
    margin: 0 1rem;
    padding: 1rem;
  }

  &--news {
    height: 100%;
    width: 100%;
  }

  &--form-navigation {
    background: var(--color--background--button);
    border: solid 1px var(--color--border);
    border-radius: 4px 4px 0 0;
    color: var(--color--font--gray);
    cursor: pointer;
    font-size: 2rem;
    line-height: 1.5;
    padding: 0.5rem;

    &.selected {
      background: var(--color--background--form);
      border-bottom: none;
      color: var(--color--white);
    }

    &:focus {
      background: var(--color--background--button--hover);
      outline: none;
    }

    &.login {
      grid-area: login;
      margin-left: 1rem;
    }

    &.signup {
      grid-area: signup;
      margin-right: 1rem;
    }
  }

  &--title {
    margin-top: 10vh;

    & > p {
      font-size: 3rem;
      margin: 1rem;
      text-align: center;
    }
  }

  &--sponsor {
    grid-area: sponsor;
  }

  &--footer {
    grid-area: footer;
  }
}
