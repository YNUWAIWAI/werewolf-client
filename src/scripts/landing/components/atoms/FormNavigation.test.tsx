import * as React from 'react'
import {Language, getMessages} from '../../../../i18n/landing'
import FormNavigation from './FormNavigation'
import {initRenderer} from '../../../../../tools/intl-enzyme-test-helper'

const {mountWithIntl} = initRenderer(Language.en, getMessages(Language.en))

describe('render', () => {
  test('isSignup={true}', () => {
    const handleClick = jest.fn()
    const wrapper = mountWithIntl(
      <FormNavigation
        handleClick={handleClick}
        isSignup
      />
    )

    expect(wrapper.text()).toBe('Login')
  })
  test('isSignup={false}', () => {
    const handleClick = jest.fn()
    const wrapper = mountWithIntl(
      <FormNavigation
        handleClick={handleClick}
        isSignup={false}
      />
    )

    expect(wrapper.text()).toBe('Signup')
  })
})
