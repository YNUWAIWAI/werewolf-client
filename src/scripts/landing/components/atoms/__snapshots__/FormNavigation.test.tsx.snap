// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render isSignup={false} 1`] = `"<button class=\\"la--form-navigation\\">Signup</button>"`;

exports[`render isSignup={true} 1`] = `"<button class=\\"la--form-navigation\\">Login</button>"`;
